WordPress/src/main/java/org/wordpress/android/ui/plugins/PluginDetailActivity.java

```diff
@@ -1066,4 +1113,238 @@ private boolean isPluginStateChangedSinceLastConfigurationDispatch() {
+            } else {
+                mAutomatedTransferProgressDialog.setProgress(event.currentStep * 100 / event.totalSteps);
+                mHandler.postDelayed(new Runnable() {
+                    @Override
+                    public void run() {
+                        // Let's wait a second before checking the status again
+                        mDispatcher.dispatch(SiteActionBuilder.newCheckAutomatedTransferStatusAction(mSite));
+                    }
+                }, 1000);
+            }
+        }
+    }
+
+    /**
+     * Once the Automated Transfer is completed, we'll trigger a fetch for the site since it'll become a Jetpack site.
+     * Whenever the site is updated we update `mSite` property. If the Automated Transfer progress dialog is
+     * showing and we make sure that the updated site has the correct `isAutomatedTransfer` flag, we fetch the site
+     * plugins so we can refresh this page.
+     *
+     * Check out `onPluginDirectoryFetched` for the last step of a successful Automated Transfer.
+     */
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onSiteChanged(OnSiteChanged event) {
+        // We should be safe to ignore the errors
+        if (isFinishing() && event.isError()) {
+            return;
+        }
+        mSite = mSiteStore.getSiteBySiteId(mSite.getSiteId());
+
+        // We try to fetch the site after Automated Transfer is completed so that we can fetch its plugins. If
+        // we are still showing the AT progress and the site is AT site, we can continue with plugins fetch
+        if (mIsShowingAutomatedTransferProgress && mSite.isAutomatedTransfer()) {
+            mDispatcher.dispatch(PluginActionBuilder.newFetchPluginDirectoryAction(new PluginStore
+                    .FetchPluginDirectoryPayload(PluginDirectoryType.SITE, mSite, false)));
+        }
+    }
+
+    /**
+     * Completing an Automated Transfer will trigger a site fetch which then will trigger a fetch for the site plugins.
+     * We'll complete the Automated Transfer if the progress dialog is showing and only update the plugin and the views
+     * if it's not.
+     *
+     * This event is unlikely to happen outside of Automated Transfer process, and it is even less likely that the views
+     * will need to be updated because of it, but they are both still possible and we try to handle it with a refresh.
+     */
+    @SuppressWarnings("unused")
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public void onPluginDirectoryFetched(OnPluginDirectoryFetched event) {
+        // We should be safe to ignore any errors related to plugin directory fetches since the only time we trigger it
+        // from this view is during AT progress and it should hopefully always succeed and we can't really do much about
+        // it anyway.
+        if (isFinishing() || event.isError()) {
+            return;
```

Comment:
If there's an error, shouldn't we hide the AT progress dialog? Otherwise it could be left showing and the user can't do anything to get rid of it (outside of killing the app).

Reply:
Yes, I am planning to do that, however I wasn't sure if it could have somehow killed the current AT process and wanted to make sure I handled it. I should have left a todo for that.